navigation/navigation-common/src/jbMain/kotlin/androidx/navigation/NavDestination.jb.kt

```diff
@@ -124,7 +124,7 @@ public actual open class NavDestination actual constructor(
     }
 
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    public actual open fun matchDeepLink(route: String): DeepLinkMatch? {
+    public actual fun matchDeepLink(route: String): DeepLinkMatch? {
```

Comment:
I know this place is a mess, but you're removing an ability to override it from public. Is it expected? It's restricted to `LIBRARY_GROUP`, so probably it's ok 🤔

Reply:
yes. I synchronized the API with the android part. there was a mistake. The change fixes the problem with a nested navigation
https://github.com/JetBrains/compose-multiplatform-core/blob/778cf002523049dfcf792198d3c223bbe3c181e2/navigation/navigation-common/src/androidMain/kotlin/androidx/navigation/NavDestination.android.kt#L355

Reply:
it should fix https://youtrack.jetbrains.com/issue/CMP-6689
I will check